servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancer.java

```diff
@@ -417,14 +454,16 @@ public void onComplete() {
                         // Failure in selection could be temporary, hence add it to the queue and be consistent
                         // with the fact that select failure does not close a connection.
                         return newCnx.closeAsync().concat(failed(new ConnectionRejectedException(
-                                "Newly created connection " + newCnx + " rejected by the selection filter.")));
+                                "Newly created connection " + newCnx + " for " + targetResource
+                                        + " was rejected by the selection filter.")));
                     }
                     if (host.addConnection(newCnx)) {
                         return succeeded(newCnx);
                     }
-                    return newCnx.closeAsync().concat(this.usedHosts == CLOSED_LIST ? failedLBClosed() :
+                    return newCnx.closeAsync().concat(this.usedHosts == CLOSED_LIST ? failedLBClosed(targetResource) :
                             failed(new ConnectionRejectedException(
-                                    "Failed to add newly created connection for host: " + host)));
+                                    "Failed to add newly created connection for " + targetResource
+                                            + " for host: " + host)));
```

Comment:
Please note this part of the exception has not changed in this PR.

Answering the question though, I believe it's essential to capture the state of the host and current number of connections upon the rejection. Not including that information might leave the users in the dark while debugging. Especially as we should only see rejections happen when the host is in closed state. Not including the state would leave some questions unanswered.

Reply:
Fair, let's keep both then (`newCtx` and `host`) in this exception message because it's also important to know what connection id was rejected.

Reply:
Because `Host#toString()` already says that it's the `Host`, consider:
```
"Failed to add newly created connection " + newCnx + " for " + targetResource + " for " + host
```